libraries/AP_Vehicle/AP_Vehicle.h

```diff
@@ -225,6 +225,20 @@ class AP_Vehicle : public AP_HAL::HAL::Callbacks {
     // returns true if vehicle is in the process of taking off
     virtual bool is_taking_off() const { return false; }
 
+    // landed state enumeration
+    // not strictly required, but should mirror MAV_LANDED_STATE enum
+    enum class LandedState {
+        Undefined = 0,      // landed state unknown/not available
+        OnGround = 1,       // vehicle is landed (on ground)
+        InAir = 2,          // vehicle is in air
+        TakeOff = 3,        // vehicle currently taking off
+        Landing = 4,        // vehicle currently landing
+        Last_LandedState   // place new values before this
```

Comment:
We don't want this one.  So there's the strict 1:1 mapping broken.  But no need for this here AFAICS.